tools/read_mutation.hh

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2021-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <filesystem>
+#include <functional>
+#include <string_view>
+#include <optional>
+
+#include "tools/sstable_manager_service.hh"
+#include "schema/schema_fwd.hh"
+#include "types/types.hh"
+#include "reader_permit.hh"
+
+mutation_opt read_mutation_from_table_offline(sharded<sstable_manager_service>& sst_man,
+                                              reader_permit permit,
+                                              std::filesystem::path table_path,
+                                              std::string_view keyspace,
+                                              std::function<schema_ptr()> table_schema,
+                                              data_value primary_key,
+                                              std::optional<data_value> clustering_key);
```

Comment:
The last two parameters are specific to the current use-case: reading from the schema tables, which all have the same primary key. Once we start using this method to read other tables (`system.tablets`), these parameters become incorrect/obsolete. So best to first refactor this method to make it table-independent: pass `partition_key` and `clustering_key`, which is produced by the callers, who know how to generate these based on the table they are working with.

Reply:
> The last two parameters are specific to the current use-case: reading from the schema tables, which all have the same primary key. Once we start using this method to read other tables (`system.tablets`), these parameters become incorrect/obsolete. 

i am confused. when reading `system.tablets` , this very function is used, like
```c++
    auto mut = read_mutation_from_table_offline(sst_man,
                                                permit,
                                                tablets_table_directory,
                                                db::system_keyspace::NAME,
                                                db::system_keyspace::tablets,
                                                data_value(table_id),
                                                {});
```
and the `primary_key` is neither incorrect nor obsolete. it is used to pass the `table_id`. 

> So best to first refactor this method to make it table-independent: pass `partition_key` and `clustering_key`, which is produced by the callers, who know how to generate these based on the table they are working with.

it is tablet-independent. `partition_key` and `clustering_key` are passed in. also i was following the suggestion at https://github.com/scylladb/scylladb/pull/20002#discussion_r1703112457. what do you propose exactly ?

Reply:
Actually, these two parameters will work fine for any table which has a single-component partition key. So we can leave it like this. Sorry for the noise, for some reason I thought this only works for partition keys with a single `text` column.